src/player.cpp

```diff
@@ -5268,38 +5269,38 @@ void player::suffer()
         sounds::sound( pos(), 10, sounds::sound_t::movement, _( "BZZZZZ" ) );
     }
 
-    bool wearing_shoes = is_wearing_shoes( side::LEFT ) || is_wearing_shoes( side::RIGHT );
-    if( has_trait( trait_ROOTS3 ) && g->m.has_flag( "PLOWABLE", pos() ) && !wearing_shoes ) {
-        if( one_in( 100 ) ) {
-            add_msg_if_player( m_good, _( "This soil is delicious!" ) );
-            if( get_hunger() > -20 ) {
-                mod_hunger( -2 );
-                // absorbs kcal directly
-                mod_stored_nutr( -2 );
-            }
-            if( get_thirst() > -20 ) {
-                mod_thirst( -2 );
-            }
-            mod_healthy_mod( 10, 50 );
-            // No losing oneself in the fertile embrace of rich
-            // New England loam.  But it can be a near thing.
-            /** @EFFECT_INT decreases chance of losing focus points while eating soil with ROOTS3 */
-            if( ( one_in( int_cur ) ) && ( focus_pool >= 25 ) ) {
-                focus_pool--;
-            }
-        } else if( one_in( 50 ) ) {
-            if( get_hunger() > -20 ) {
-                mod_hunger( -1 );
-                // absorbs kcal directly
-                mod_stored_nutr( -1 );
-            }
-            if( get_thirst() > -20 ) {
-                mod_thirst( -1 );
-            }
-            mod_healthy_mod( 5, 50 );
-        }
+    bool has_leaves = has_trait( trait_LEAVES );
+    bool has_roots = ( has_trait( trait_ROOTS2 ) || has_trait( trait_ROOTS3 ) );
+    bool is_rooted = has_effect( effect_rooted );
+    bool is_rooting = has_activity( activity_id( "ACT_ROOT" ) );
+    bool is_uprooting = has_activity( activity_id( "ACT_UPROOT" ) );
+
+    if( is_rooted ) {
+        rooted();
+    }
+    // SANITY IS FOR THE WEAK
+
+    if( has_leaves || has_roots ) {
+        plant_nutrition();
+        // This was getting complicated, so it now gets its own method.
     }
 
+    if( has_active_mutation( trait_ROOTS2 ) && !is_rooted && !is_rooting ) {
+        deactivate_mutation( trait_ROOTS2 );
+    }
+    if( has_active_mutation( trait_ROOTS3 ) && !is_rooted && !is_rooting ) {
+        deactivate_mutation( trait_ROOTS3 );
+    }
```

Comment:
There's no need to check for having active mutations, so these two checks above could be rewritten as

```
if( !is_rooted && !is_rooting ) {
    deactivate_mutation( trait_ROOTS2 );
    deactivate_mutation( trait_ROOTS3 );
}
```

Reply:
Can I deactivate a mutation the player doesn't have safely?

Reply:
Yeah. If you have the mutation in question, then it will be deactivated. If you don't have the mutation, nothing will happen.

Reply:
Research has pointed me to an answer of no, it will cause both mutation to appear in the menu.

Reply:
Well, ok.